File: frontend/app/vo/messages/MessageVo.ts
/**
 * 백엔드 com.jsh.aidesk.serverapi.messages.vo.MessageItemRsVo 대응.
 */
export type MessageStatus = 'sent' | 'delivered' | 'replied' | 'failed';

export interface AttachmentRef {
  attachmentId: string;
  originalFilename: string;
  contentType: string;
  sizeBytes: number;
}

export interface MessageItem {
  messageId: string;
  fromAgentId: string;
  fromAgentName: string;
  toAgentId: string;
  toAgentName: string;
  content: string;
  replyToMessageId: string | null;
  status: MessageStatus;
  errorReason: string | null;
  createdAt: string;          // ISO 8601
  deliveredAt: string | null;
  readAt: string | null;
  repliedAt: string | null;
  attachments?: AttachmentRef[];
}

export interface MessageCreateRequest {
  fromAgentId: string;
  toAgentId: string;
  content: string;
  replyToMessageId?: string;
  attachmentIds?: string[];
}

export interface AttachmentUploadResponse {
  attachmentId: string;
  originalFilename: string;
  contentType: string;
  sizeBytes: number;
}

File: backend/src/main/java/com/jsh/aidesk/serverapi/messages/vo/MessageVo.java
package com.jsh.aidesk.serverapi.messages.vo;

import java.time.OffsetDateTime;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class MessageVo {

    private String messageId;
    private String fromAgentId;
    private String toAgentId;
    private String content;
    private String replyToMessageId;
    private String rootMessageId;
    private Integer hopCount;
    private String status;
    private String errorReason;
    private Integer retryCount;
    private OffsetDateTime lastAttemptAt;
    private OffsetDateTime createdAt;
    private OffsetDateTime deliveredAt;
    private OffsetDateTime readAt;
    private OffsetDateTime repliedAt;
}
